### ISSUE TITLE
tests should detect unity installations on linux

### ISSUE THREAD

**Issue by heeen** [microsoft/Microsoft.Unity.Analyzers#264]
#### Bug description

unit tests fail to run on linux without warning about unsupported operating system

- Version of analyzers assembly:  79a79a9
- Analyzer rule: most of them
- Error (exception message, type, and callstack where applicable): 

```
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:01.63]     Microsoft.Unity.Analyzers.Tests.PropertyDrawerOnGUITests.TestBaseOnGUIOverride [FAIL]
[xUnit.net 00:00:01.64]     Microsoft.Unity.Analyzers.Tests.SetLocalPositionAndRotationTests.DoNotFailWithInterfaceMembers [FAIL]
[xUnit.net 00:00:01.64]     Microsoft.Unity.Analyzers.Tests.Vector2MathTests.Multiple [FAIL]
[xUnit.net 00:00:01.64]     Microsoft.Unity.Analyzers.Tests.LoadAttributeMethodTests.InitializeOnLoadMethodTest [FAIL]
[xUnit.net 00:00:01.64]     Microsoft.Unity.Analyzers.Tests.ThrowExpressionSuppressorTests.SuppressThrowExpressionWithUnityObjectsCodeBlock [FAIL]
  Failed Microsoft.Unity.Analyzers.Tests.PropertyDrawerOnGUITests.TestBaseOnGUIOverride [627 ms]
  Error Message:
   System.InvalidOperationException : Sequence contains no elements
  Stack Trace:
     at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Microsoft.Unity.Analyzers.Tests.UnityPath.FirstInstallation() in /home/florian/src-misc/roslyn-analyzers/Microsoft.Unity.Analyzers/src/Microsoft.Unity.Analyzers.Tests/UnityPath.cs:line 22
   at Microsoft.Unity.Analyzers.Tests.DiagnosticVerifier.UnityAssemblies()+MoveNext() in /home/florian/src-misc/roslyn-analyzers/Microsoft.Unity.Analyzers/src/Microsoft.Unity.Analyzers.Tests/Infrastructure/DiagnosticVerifier.cs:line 329
   at System.Linq.Enumerable.Aggregate[TSource,TAccumulate](IEnumerable`1 source, TAccumulate seed, Func`3 func)
   at Microsoft.Unity.Analyzers.Tests.DiagnosticVerifier.CreateProject(AnalyzerVerificationContext context, String[] sources) in /home/florian/src-misc/roslyn-analyzers/Microsoft.Unity.Analyzers/src/Microsoft.Unity.Analyzers.Tests/Infrastructure/DiagnosticVerifier.cs:line 374
   at Microsoft.Unity.Analyzers.Tests.DiagnosticVerifier.GetDocuments(AnalyzerVerificationContext context, String[] sources) in /home/florian/src-misc/roslyn-analyzers/Microsoft.Unity.Analyzers/src/Microsoft.Unity.Analyzers.Tests/Infrastructure/DiagnosticVerifier.cs:line 311
   at Microsoft.Unity.Analyzers.Tests.DiagnosticVerifier.GetSortedDiagnosticsAsync(AnalyzerVerificationContext context, String[] sources, DiagnosticAnalyzer analyzer, DiagnosticResult[] expected) in /home/florian/src-misc/roslyn-analyzers/Microsoft.Unity.Analyzers/src/Microsoft.Unity.Analyzers.Tests/Infrastructure/DiagnosticVerifier.cs:line 210
   at Microsoft.Unity.Analyzers.Tests.DiagnosticVerifier.VerifyDiagnosticsAsync(AnalyzerVerificationContext context, String[] sources, DiagnosticAnalyzer analyzer, DiagnosticResult[] expected) in /home/florian/src-misc/roslyn-analyzers/Microsoft.Unity.Analyzers/src/Microsoft.Unity.Analyzers.Tests/Infrastructure/DiagnosticVerifier.cs:line 81
   at Microsoft.Unity.Analyzers.Tests.PropertyDrawerOnGUITests.TestBaseOnGUIOverride() in /home/florian/src-misc/roslyn-analyzers/Microsoft.Unity.Analyzers/src/Microsoft.Unity.Analyzers.Tests/PropertyDrawerOnGUITests.cs:line 124
--- End of stack trace from previous location ---
```

#### To Reproduce
Steps or code to reproduce the behavior:
1. be on linux
2. build project
3. try to run tests

#### Expected behavior
it should detect installations on linux and run tests
-or-
it should warn about linux being unsupported
